### PR TITLE
Add Cloud Functions & Firestore tutorial

### PR DESCRIPTION
**Pull request by zeroasterisk**
- [x] - Editorial review - @ToddKopriva  
- [x] - Final production check - @ToddKopriva

### CODE REVIEW

File: tutorials/cloud-functions-firestore/index.md [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+return the document if found.
+
+NOTE: you probably want more input sanitation for a production application.
+
+## Deploying the Function
+
+You now have a `package.json` file listing your dependencies
+and you have an `index.js` file which will listen for a HTTP trigger.
+
+We want to make google automatically run that code for us, when needed.
```
Comment (ToddKopriva) [inline tutorials/cloud-functions-firestore/index.md]: Capitalize 'Google'. Also, which part of Google, exactly?

File: tutorials/cloud-functions-firestore/index.md [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+[local emulator for Google Cloud Functions][emulator].
+
+[sdk]: https://github.com/GoogleCloudPlatform/cloud-functions-emulator
+
+    npm install -g @google-cloud/functions-emulator
+    export GOOGLE_APPLICATION_CREDENTIALS=/Users/myname/.cred/myserviceaccount.json
+    functions start
+    functions deploy main --trigger-http
+
+In this case, `main` is the name of the function you want to trigger in your code, triggered by a HTTP request.
```
Comment (ToddKopriva) [inline tutorials/cloud-functions-firestore/index.md]: 'a HTTP'
->
'an HTTP'

File: tutorials/cloud-functions-firestore/index.md [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+
+NOTE that our function code added `created`.
+
+### Deploy the Function to GCP Functions
+
+This is very easy thanks to the gcloud CLI.
+
+    gcloud functions deploy main --trigger-http
+
+In this case, `main` is the name of the function you want to trigger in your code, triggered by a HTTP request.
```
Comment (ToddKopriva) [inline tutorials/cloud-functions-firestore/index.md]: 'a HTTP'
->
'an HTTP'

File: tutorials/cloud-functions-firestore/index.md [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+    name: projects/cloud-functions-firestore/locations/us-central1/functions/main
+    runtime: nodejs6
+    serviceAccountEmail: cloud-functions-firestore@appspot.gserviceaccount.com
+    sourceUploadUrl: https://storage.googleapis.com/gcf-upload-us-central1-0000000000.zip?GoogleAccessId=service-...
+    status: ACTIVE
+    timeout: 60s
+    updateTime: '2018-09-15T01:25:59Z'
+    versionId: '1'
+
+## Testing the Deployed Function in Production
```
Comment (ToddKopriva) [inline tutorials/cloud-functions-firestore/index.md]: We use sentence case for headings, not title case.

File: tutorials/cloud-functions-firestore/index.md [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+## Iterate
+
+This is a very short cycle of `[code, deploy, test]`
+so you should be able to iterate rapidly.
+
+When you deploy, you overwrite the current version, at the function's URL *(blue/green deployment in the background, can take a few seconds to switch over)*.
+
+If you need to access a previously deployed version, you can append `/revisions/${REVISION}` with the value of the `versionId` that deploy gives you back.
+
+ There are lot of other settings available, review docs and help
```
Comment (ToddKopriva) [inline tutorials/cloud-functions-firestore/index.md]: This is two sentences. Replace the comma with a period. End the second sentence with a colon, to lead into the command that follows.

File: tutorials/cloud-functions-firestore/index.md [hunk 6]
```diff
@@ -0,0 +1,256 @@
+---
+title: Use Google Cloud Firestore with Functions
+description: Learn how to use Google Cloud Functions to store and retrieve data with Google Cloud Firestore.
+author: zeroasterisk
+tags: Cloud Functions, Cloud Datastore, Cloud Firestore
+date_published: 2018-09-30
```
[the remark this answers is not in the document]
Comment (zeroasterisk) [reply tutorials/cloud-functions-firestore/index.md]: updated, but you are welcome to change as needed

File: tutorials/cloud-functions-firestore/index.md [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+[node]: https://nodejs.org/en/
+
+The sample Cloud Function is triggered by a web request,
+which you can simulate with `curl`.
+
+## Prerequisites
+
+1.  Create a project in the [Google Cloud Platform Console][console].
+1.  Enable billing for your project.
+1.  Enable Functions and Firebase *(under **APIs > Enable APIs and Services**)*
```
[the remark this answers is not in the document]
Comment (zeroasterisk) [reply tutorials/cloud-functions-firestore/index.md]: fixed (thanks!)

File: tutorials/cloud-functions-firestore/index.md [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+1.  Install the [Google Cloud SDK][sdk].
+
+[console]: https://console.cloud.google.com/
+[sdk]: https://cloud.google.com/sdk/
+
+Configure the gcloud CLI for your project_id.
```
[the remark this answers is not in the document]
Comment (zeroasterisk) [reply tutorials/cloud-functions-firestore/index.md]: I ended up searching around in the repo to see how others did this.  I ended up copying this pattern: 
https://raw.githubusercontent.com/GoogleCloudPlatform/community/9003201b20699aa25eb8586108d13be61e3b4c94/tutorials/cloud-functions-twilio-voice-record/index.md

But if you prefer yours, I'll switch to that. Just let me know.

File: tutorials/cloud-functions-firestore/index.md [hunk 9]
```diff
@@ -0,0 +1,256 @@
...
+
+    gcloud config set project <projectid>
+
+For the rest of this guide, I'll be using `cloud-functions-firestore` as the project_id, as well as the firestore collection.
```
[the remark this answers is not in the document]
Comment (zeroasterisk) [reply tutorials/cloud-functions-firestore/index.md]: ok - will make several corrections for this

File: tutorials/cloud-functions-firestore/index.md [hunk 10]
```diff
@@ -0,0 +1,256 @@
...
+
+## Preparing the Cloud Function
+
+For this Function, we are going to either
```
[the remark this answers is not in the document]
Comment (zeroasterisk) [reply tutorials/cloud-functions-firestore/index.md]: using Cloud Function

File: tutorials/cloud-functions-firestore/index.md [hunk 11]
```diff
@@ -0,0 +1,256 @@
...
+store a basic POST payload as a document in Firestore
+or retrieve a document from Firestore by id.
```
[the remark this answers is not in the document]
Comment (zeroasterisk) [reply tutorials/cloud-functions-firestore/index.md]: done

File: tutorials/cloud-functions-firestore/index.md (hunk 1 above)
Comment (zeroasterisk) [reply tutorials/cloud-functions-firestore/index.md]: using "Google Cloud Functions"

File: tutorials/cloud-functions-firestore/index.md (hunk 2 above)
Comment (zeroasterisk) [reply tutorials/cloud-functions-firestore/index.md]: fixed

File: tutorials/cloud-functions-firestore/index.md (hunk 3 above)
Comment (zeroasterisk) [reply tutorials/cloud-functions-firestore/index.md]: fixed

File: tutorials/cloud-functions-firestore/index.md (hunk 4 above)
Comment (zeroasterisk) [reply tutorials/cloud-functions-firestore/index.md]: fixed

File: tutorials/cloud-functions-firestore/index.md (hunk 5 above)
Comment (zeroasterisk) [reply tutorials/cloud-functions-firestore/index.md]: done

**Comment by ToddKopriva** [comment]
published:
https://cloud.google.com/community/tutorials/cloud-functions-firestore

**Pull request merged**